modules/backend/widgets/Form.php

```diff
@@ -1069,13 +1069,21 @@ protected function getFieldValue($field)
             $field = $this->allFields[$field];
         }
 
-        $defaultValue = !$this->model->exists
+        $defaultValue = $this->shouldFetchDefaultValues()
             ? $field->getDefaultFromData($this->data)
             : null;
 
         return $field->getValueFromData($this->data, $defaultValue);
     }
 
+    /**
+     * Checks if default values should be taken from data.
+     * This should be done when model exists or when explicitly configured
+     */
+    protected function shouldFetchDefaultValues() {
+        return !$this->model->exists || object_get($this->config, 'shouldFetchDefaultValues');
```

Comment:
If we're using a config value for this it should support being explicitly set to false and having that override `!$this->model->exists`